src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -1413,7 +1416,112 @@ private IBlockOperation CreateBoundBlockOperation(BoundBlock boundBlock)
+        }
+
+        private ConditionalOperation GenerateNullCheckForParameter(ParameterSymbol parameter, SyntaxNode syntax)
+        {
+            Optional<object> constantValue = default(Optional<object>);
+            var compilation = (CSharpCompilation)_semanticModel.Compilation;
+            IOperation conditionOp;
+            if (parameter.Type.IsNullableType())
+            {
+                conditionOp = new UnaryOperation(
+                    UnaryOperatorKind.Not,
+                    new InvocationOperation(
+                        targetMethod: ((IMethodSymbol)compilation.GetSpecialTypeMember(SpecialMember.System_Nullable_T_get_HasValue)).Construct(parameter.Type),
+                        instance: new ParameterReferenceOperation(parameter, _semanticModel, syntax, parameter.Type, constantValue, isImplicit: true),
+                        isVirtual: false,
+                        arguments: ImmutableArray<IArgumentOperation>.Empty,
+                        _semanticModel,
+                        syntax,
+                        compilation.GetSpecialType(SpecialType.System_Boolean),
+                        constantValue,
+                        isImplicit: true),
+                    isLifted: false,
+                    isChecked: false,
+                    operatorMethod: null,
+                    _semanticModel,
+                    syntax,
+                    compilation.GetSpecialType(SpecialType.System_Boolean),
+                    constantValue,
+                    isImplicit: true);
+            }
+            else
+            {
+                conditionOp = new BinaryOperation(BinaryOperatorKind.Equals,
+                    new ParameterReferenceOperation(parameter, _semanticModel, syntax, parameter.Type, constantValue, isImplicit: true),
+                    new LiteralOperation(_semanticModel, syntax, parameter.Type, ConstantValue.Null, isImplicit: true),
+                    isLifted: false,
+                    isChecked: false,
+                    isCompareText: false,
+                    operatorMethod: null,
+                    unaryOperatorMethod: null,
+                    _semanticModel,
+                    syntax,
+                    compilation.GetSpecialType(SpecialType.System_Boolean),
+                    constantValue,
+                    isImplicit: true);
+            }
+
+            var argumentNullExceptionMethodSymbol = (IMethodSymbol)compilation.GetWellKnownTypeMember(WellKnownMember.System_ArgumentNullException__ctorString);
+            var argumentNullExceptionType = compilation.GetWellKnownType(WellKnownType.System_ArgumentNullException);
+            var argumentNullExceptionObject = new ObjectCreationOperation(
+                        constructor: argumentNullExceptionMethodSymbol,
+                        initializer: new ObjectOrCollectionInitializerOperation(
```

Comment:
>new ObjectOrCollectionInitializerOperation [](start = 37, length = 42)

Do we need an initializer here or is `null` sufficient?

Reply:
Should be `null`

---
In reply to: [305538212](https://github.com/dotnet/roslyn/pull/37381#discussion_r305538212) [](ancestors = 305538212)